lib/Basics/StringUtils.cpp

```diff
@@ -1406,6 +1414,119 @@ std::string soundex(char const* src, size_t const len) {
   return result;
 }
     
+    
+template<typename InputType>
+inline bool eq(InputType const& c1, InputType const& c2) {
+    return c1 == c2;
+}
+
+template<typename InputType, typename LengthType>
+LengthType impl(InputType const* lhs,
+                               InputType const* rhs,
+                               LengthType lhsSize,
+                               LengthType rhsSize) {
+    assert(lhsSize >= rhsSize);
```

Comment:
in ArangoDB, we are not using `assert(x)` directly. All assertions in ArangoDB should use `TRI_ASSERT(x)` instead.